linalg/batched/gpu_blas.cpp

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2010-2024, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "gpu_blas.hpp"
+#include "../../general/forall.hpp"
+
+#ifdef MFEM_USE_CUDA_OR_HIP
+
+#if defined(MFEM_USE_CUDA)
+#define MFEM_cu_or_hip(stub) cu##stub
+#define MFEM_CU_or_HIP(stub) CU##stub
+#elif defined(MFEM_USE_HIP)
+#define MFEM_cu_or_hip(stub) hip##stub
+#define MFEM_CU_or_HIP(stub) HIP##stub
+#endif
```

Comment:
Minor comment, but these could be moved to general as they get used a few times in MFEM.

Reply:
Yeah, these are also used currently in sparsemat.cpp. Right now they are only in cpp files, so that these definitions don't escape out of the library. We could put them in general in a hpp file that we are careful only to include in internal cpp files.